src/python/pants/backend/python/rules/coverage.py

```diff
@@ -295,21 +297,28 @@ async def generate_coverage_reports(
             )
         )
     results = await MultiGet(Get(ProcessResult, PexProcess, process) for process in pex_processes)
+    result_snapshots = await MultiGet(Get(Snapshot, Digest, r.output_digest) for r in results)
```

Comment:
this pattern is going to repeat in a lot of places.
We should have a way (rule) to get a Snapshot from a PexProcess.

Reply:
> We should have a way (rule) to get a Snapshot from a PexProcess.

We do have a rule: `await Get(Snapshot, Digest, r.output_digest)`. That is the canonical way to uplift a digest in the engine, as documented in https://www.pantsbuild.org/v2.0/docs/rules-api-file-system. While a little boilerplatey, this line is the canonical way of doing things. 

We should keep it this way because a) with so many APIs, it's valuable to keep things simple and consistent so that they're predictable, and b) I don't even think it'd be possible to improve on this. You'd need to do something like this, which is worse imo:

```
result_snapshots = await Get(Snapshots, SnapshotsFromProcessResults(results))
```

Where both `Snapshots` and `SnapshotsFromProcessResults` are new types that would have to be imported.